### DISCUSSION TITLE
CSS not working?  what is going on?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#36777] [Help] [answered]
So I am new to Next.js and working on migrating a site.

I am trying to center a picture and a button (button from bootstrap, yes, I am using bootstrap too) and it isn't taking the css given to them.

While inspecting the item I see something like element.style{} with a bunch of values in it.  Where did it come from?  And how do I change it? 

See attached picture. 

![css1](https://user-images.githubusercontent.com/93617433/167435992-0433ccd2-00cd-43a8-99d7-5e11eb3282a6.PNG)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Are you using `next/image`? 

It has been designed by a bunch of engineers to have those styles, aka, best practices. Since they are `inlined` they have higher specificity that anything you may add via class names.

What it is usually recommended is to target the wrapper around it. You can read more here, https://nextjs.org/docs/basic-features/image-optimization#styling.

## Edit

If migration and using next/image at the same time is blocking you, then you should migrate to next.js first, and, only then, as a next step migrate the images and scripts. Always taking a moment to understand these APIs.

**Comment by Elindo586** [comment]
![snip2](https://user-images.githubusercontent.com/93617433/167456040-357797b9-bc85-40b2-8974-3759c9949955.PNG)
@icyJoseph 

So, when I say "migrate", really means just copying and pasting page by page from a CRA code to an npx create next file system.   At the end I will end up with 2 codes from different platforms... 

In this case I haven't figured out how to add the current cra styles to the next.js code.

I have this for example:
(See attached pic)
from page:
 ```
<span className="rexrothpump">
              <Image
                src="/../public/images/rexrothpump-1.png"
                className="rexrothpump"
                alt="piston-pump"
                width={263}
                height={191}
              />
            </span>

```
from  global CSS

```
.rexrothpump,
{
  display: block;
  margin-left: auto;
  margin-right: auto;
}
```

The console is showing the styles, but not taking them...

I also have this NormalizingError?  Is that something I should fix? if so, how do I do that?

```
NormalizeError: Requested and resolved page mismatch: /../public/images/rexrothpump-1.png /public/images/rexrothpump-1.png
    at Object.normalizePagePath (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\shared\lib\page-path\normalize-page-path.js:14:15)       
    at DevServer.hasPage (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\dev\next-dev-server.js:335:54)
    at DevServer._beforeCatchAllRender (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\dev\next-dev-server.js:359:28)
    at async Object.fn (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\dev\next-dev-server.js:610:21)
    at async Router.execute (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\router.js:226:36)
    at async DevServer.run (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\base-server.js:606:29)
    at async DevServer.run (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node
![snip2](https://user-images.githubusercontent.com/93617433/167455778-1732f44e-14aa-45a6-b8c5-dc1cd728e281.PNG)
_modules\next\dist\server\dev\next-dev-server.js:450:20)
    at async DevServer.handleRequest (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\base-server.js:321:20)
    at async imageOptimizer (C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\image-optimizer.js:241:13)
    at async C:\Users\Edgar\Desktop\Stuff\Acode\nextprojecta\node_modules\next\dist\server\next-server.js:179:72

```

**Reply by icyJoseph** [reply]
You don't need to resolve the path to the `public` directory. If the images are in `<rootDir>/public`, then you can just do:

```jsx
<span className="rexrothpump">
             <Image
               src="/images/rexrothpump-1.png"
               className="rexrothpump"
               alt="piston-pump"
               width={263}
               height={191}
             />
</span>
```

**Comment by Elindo586** [comment]
![snip2](https://user-images.githubusercontent.com/93617433/167456081-058f6f32-24a8-48e4-94ed-5efaf9d4bbe2.PNG)

**Reply by icyJoseph** [reply]
You have to see that `next/image` is rendering more than just an image, it wraps it with a span, and such.

In this case, you cannot win a specificity war against inline styles, but you could wrap the `next/image`, with a `div` (or figure), and style that according to your needs. 

The inlined styles are best practices, and even necessary to drive the value proposition of using `next/image`.

It is possible to style the image further, by passing a class name that doesn't touch those properties. 

One thing to notice is that, in your case, the inlined styles apply the same, `display: block;` and `margin : auto;`, although yours targeted only `left/right` margins.

**Comment by Elindo586** [comment]
This code below worked.

For some reason it took the bootstrap code, but didn't take my code


```
 <div className="d-flex justify-content-center">
              <Image
                src="/../public/images/rexrothpump-1.png"
                alt="piston-pump"
                width={263}
                height={191}
              />
            </div>
```

**Reply by icyJoseph** [reply]
Don't forget that you don't need to target the `public` folder directly:

> Next.js can serve static files, like images, under a folder called public in the root directory. Files inside public can then be referenced by your code starting from the base URL (/).

> For example, if you add an image to `public/me.png`, the following code will access the image:

```jsx
import Image from 'next/image'

function Avatar() {
  return <Image src="/me.png" alt="me" width="64" height="64" />
}

export default Avatar
```

**Reply by Elindo586** [reply]
Thanks a million..

That might have been part of the problem?

Surely made a difference when targeting a pdf using the <link> tag.

**Reply by icyJoseph** [reply]
Nice! Don't forget to mark as answered, and feel free to keep on asking :)

**Comment by icyJoseph** [comment]
Did you resolve this?
